packages/lesswrong/components/hooks/useOnNavigate.tsx

```diff
@@ -20,10 +20,14 @@ const NavigationEventSender = () => {
       if (location.pathname !== lastLocation?.pathname) {
         // Don't send the callback on the initial pageload, only on post-load navigations
         if (lastLocation) {
-          void routerOnUpdate.runCallbacks({
-            iterator: {oldLocation: lastLocation, newLocation: location},
-            properties: [],
+          captureEvent("navigate", {
+            from: lastLocation.pathname,
+            to: location.pathname,
           });
+          let change: LocationChange = {oldLocation: lastLocation, newLocation: location};
+          for(let cb of [...onNavigateFunctions]) {
+            cb(change);
```

Comment:
I'm not sure this does the same thing?  Pretty sure `runCallbacks` runs each callback on the accumulator (i.e. on the results from the previous callback).  Maybe in the navigation case it doesn't matter?

Reply:
Right, `runCallbacks` has the accumulator thing, but that was actually a problem: the on-navigate callback functions _don't_ use the accumulator, and `runCallbacks` was `console.error()`ing because they returned `undefined`.